### PR TITLE
WIP: AppContext ApplicationConfigBase for #931

### LINKED ISSUE #931: Provide a way to override the global configuration file path
Need to follow through how the default value gets set on .NET Framework and see if we can match. Notably, if there is a way to configure this via environment variables we should also allow doing this and update System.Configuration to pull from AppDomainSetup.ConfigurationFile.

At worst we should move the logic down from System.Configuration for generating this name. 

Note that this value is settable. I believe that there is no way to get System.Configuration to reset after it does it's first load on the .NET Framework, but that should also be investigated.

### PR DESCRIPTION
**Pull request by newuser**
Fixes: https://github.com/dotnet/runtime/issues/931
Is it acceptable way to use directory instead of file for AppContext as there is logic at ClientConfigPaths  for configs and user.configs that doesn't belong to AppContext?

### CODE REVIEW

**Comment by jkotas** [comment]
This fix should not be touching anything in CoreLib. All changes for this fix should be in `System.Configuration.ConfigurationManager`.

File: src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs [hunk 1]
```diff
@@ -35,6 +35,7 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
```
Comment (jkotas) [inline src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs:35]: > `100644 → 100755`

This change is setting executable bit on all files you have touched. It should be undone.

**Comment by safern** [comment]
Do we need to add tests for this?

**Comment by stephentoub** [comment]
@newuser, are you still working on this, or should we close it in the meantime?  Thanks.

**Comment by newuser** [comment]
yes, close it

пн, 5 окт. 2020 г. в 19:12, Stephen Toub <notifications@github.com>:

**Comment by stephentoub** [comment]
Ok, thank you for working on it.

**Pull request closed without merging**